Prepare 3.2.0 (#298)
ChuckerTeam/chucker: CHANGELOG.md

# Change Log

## Version 3.2.0 *(2020-04-02)*

This is a new minor release with numerous internal changes.

### Summary of changes

* Chucker won't load the whole response into memory anymore, but will mutlicast it with the help of temporary files. It allows to avoid issues with OOM, like in this [issue](https://github.com/ChuckerTeam/chucker/issues/218).
This change also allows to avoid problems with Chucker consuming responses, like reported in this [issue](https://github.com/ChuckerTeam/chucker/issues/242).
* Added a red open padlock icon to clearly indicate HTTP requests in transactions list.
* Added TLS info (version and cipher suite) into `Overview` tab.
* Added ability to encode/decode URLs.
* Added RTL support.
* Switched from AsyncTasks to Kotlin coroutines.

> * Switched from Travis CI to Github Actions.

Reviewer: Move this at the bottom or prepend with an `Infra:` label as this is the least impactful change from the client point of view.

Author: I think we can remove it at all.

> * Switched to ViewBinding.

Reviewer: Maybe we can link to the doc page of the ViewBinding library? I'm not sure all the devs will be aware of this.

* Bumped targetSDK to 29.
* Greatly increased test coverage (we will add exact numbers and reports pretty soon).

### Bugfixes

> * Fixed an [issue](https://github.com/ChuckerTeam/chucker/issues/218) with OOM exceptions on big responses.
> * Fixed an [issue](https://github.com/ChuckerTeam/chucker/issues/242) with Chucker throwing exception when used as `networkInterceptor()`.

Reviewer: Maybe we should consolidate our link styles. Later in the CHANGELOG I was having links for PRs/Issues at the bottom of the doc so is easier to read when opening the RAW file:

```suggestion
* Fixed an issue [#218] with OOM exceptions on big responses.
* Fixed an issue [#242] with Chucker throwing exception when used as `networkInterceptor()`.
```

Here you're always highlighting the word `issue` so it makes hard to extract the link.

Author: I thought that just number won't get a link. Will fix it.

> * Fixed an [issue](https://github.com/ChuckerTeam/chucker/pull/240) with exception when DexGuard is used.

Reviewer: A bit generic here. I'll change with 
```suggestion
* Fixed an [issue](https://github.com/ChuckerTeam/chucker/pull/240) with HttpHeader serialisation when using obfuscation.
```

* Fixed an [issue](https://github.com/ChuckerTeam/chucker/issues/254) with response search being case-sensitive.
* Fixed an [issue](https://github.com/ChuckerTeam/chucker/issues/255) with missing search icon on Response tab.
* Fixed an [issue](https://github.com/ChuckerTeam/chucker/issues/241) with overlapping texts.

### Dependency updates

> * Updated Kotlin to 1.3.71

Reviewer: We should mention that we now depend on `kotlinx-coroutines-core` and `kotlinx-coroutines-andriod` + the version

* Updated Android Gradle plugin to 3.6.1
* Updated Room to 2.2.5
* Updated OkHttp to 3.12.10
* Updated Detekt to 1.7.1
* Updated Dokka to 0.10.1
* Updated KtLint to 9.2.1
* Updated MaterialComponents to 1.1.0

> * Updated Gradle wrapper to 6.3

Reviewer: ```suggestion
* Updated Gradle to 6.3
```

# Change Log

## Version 3.2.0 *(2020-04-02)*

This is a new minor release with numerous internal changes.

### Summary of changes

* Chucker won't load the whole response into memory anymore, but will mutlicast it with the help of temporary files. It allows to avoid issues with OOM, like in reported in [#218].
This change also allows to avoid problems with Chucker consuming responses, like reported in [#242].
* Added a red open padlock icon to clearly indicate HTTP requests in transactions list.
* Added TLS info (version and cipher suite) into `Overview` tab.
* Added ability to encode/decode URLs.
* Added RTL support.
* Switched from AsyncTasks to Kotlin coroutines.
* Switched to [ViewBinding](https://developer.android.com/topic/libraries/view-binding).
* Bumped targetSDK to 29.
* Greatly increased test coverage (we will add exact numbers and reports pretty soon).

### Bugfixes

> * Fix for [#218] with OOM exceptions on big responses.

Reviewer: The links need to be added at the bottom of the Markdown file otherwise they won't work (there are already some)

Author: Done
